pkg/cloudprovider/azure/object_store.go

```diff
@@ -198,13 +203,13 @@ func (o *ObjectStore) Init(config map[string]string) error {
 		return err
 	}
 
-	storageAccountKey, err := getStorageAccountKey(config)
-	if err != nil {
+	storageAccountKey, env, err := getStorageAccountKey(config)
+	if err != nil || env == nil {
 		return err
 	}
 
 	// 6. get storageClient and blobClient
-	storageClient, err := storage.NewBasicClient(config[storageAccountConfigKey], storageAccountKey)
+	storageClient, err := storage.NewBasicClientOnSovereignCloud(config[storageAccountConfigKey], storageAccountKey, *env)
```

Comment:
Does `NewBasicClientOnSovereignCloud` work for the public cloud env as well? The name is kind of misleading if so :)

Reply:
yes, `NewBasicClientOnSovereignCloud` work for all clouds including azure global cloud, `NewBasicClientOnSovereignCloud` function code is here: https://github.com/Azure/azure-sdk-for-go/blob/98353253a537e7e94652d100d7d78b30a4a7ee21/storage/client.go#L280-L284